Delete FileSystemAccess data between tests

szewai opened the pull request:
#### 3729059e6a040c8c168679975c9d04dab48e5a4d
<pre>
Delete FileSystemAccess data between tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=250419">https://bugs.webkit.org/show_bug.cgi?id=250419</a>
rdar://problem/104097024

Reviewed by Geoffrey Garen.

Make sure the data left by a previous test does not affect current test.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::resetStoragePersistedState):
(WebKit::NetworkProcess::clearStorage): Deleted.
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::resetStoragePersistedState):
(WebKit::NetworkStorageManager::clearStorageForTesting): Deleted.
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.h:
* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.cpp:
(WKWebsiteDataStoreResetStoragePersistedState):
(WKWebsiteDataStoreClearStorage):
* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::resetStoragePersistedState):
(WebKit::NetworkProcessProxy::clearStorage): Deleted.
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resetStoragePersistedState):
(WebKit::WebsiteDataStore::clearStorage): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):
(WTR::TestController::resetStoragePersistedState):
* Tools/WebKitTestRunner/TestController.h: Rename clearStorage() to resetStoragePersistedState() to reflect its
actual use, and introduce a new clearStorage() function to clear multiple types of data at a time.

Canonical link: <a href="https://commits.webkit.org/258818@main">https://commits.webkit.org/258818@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3553fb021bdc0511e59209e280e0a0fc9d3191c9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102869 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11987 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35895 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112122 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172344 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106832 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13006 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2895 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95118 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109789 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108643 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9987 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93183 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37629 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91832 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24715 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79363 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5440 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26129 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5587 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2583 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11604 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45621 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6068 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7334 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
```diff
@@ -345,7 +345,7 @@ void NetworkStorageManager::estimate(const WebCore::ClientOrigin& origin, Comple
 }
 
-void NetworkStorageManager::clearStorageForTesting(CompletionHandler<void()>&& completionHandler)
+void NetworkStorageManager::resetStoragePersistedState(CompletionHandler<void()>&& completionHandler)
```
Does this function guarantee removing all data from disk? At first glance it seems to change the storage mode from Persistent to BestEffort. I'm not sure whether BestEffort guarantees deletion.

szewai replied:
As in the commit message: "Rename clearStorage() to resetStoragePersistedState() to reflect its
actual use, and introduce a new clearStorage() function to clear multiple types of data at a time.", the naming is confusing.

Current clearStorageForTesting only reset persisted state.

geoffreygaren commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -1110,14 +1110,9 @@ bool TestController::resetStateToConsistentValues(const TestOptions& options, Re
-    clearDOMCaches();
-
-    resetQuota();
     clearStorage();
+    resetQuota();
+    resetStoragePersistedState();
```
It looks like we were clearing storage here already. There must be something special about persisted state that I'm not following.

szewai replied:
Ditto; it's the confusing naming.

geoffreygaren reviewed (approved):
r=me -- but I don't think I fully understood this

geoffreygaren commented on Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.cpp:
```diff
@@ -879,9 +879,26 @@ void WKWebsiteDataStoreResetQuota(WKWebsiteDataStoreRef dataStoreRef, void* cont
+    OptionSet<WebKit::WebsiteDataType> dataTypes = {
+        WebKit::WebsiteDataType::LocalStorage,
+        WebKit::WebsiteDataType::IndexedDBDatabases,
+        WebKit::WebsiteDataType::FileSystem,
```
Maybe this is the key to the change? That we now use removeData instead of clearStorage, and clearStorage didn't clear storage for real?

szewai replied:
Yes

szewai committed:
Delete FileSystemAccess data between tests
https://bugs.webkit.org/show_bug.cgi?id=250419
rdar://problem/104097024

Reviewed by Geoffrey Garen.

Make sure the data left by a previous test does not affect current test.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::resetStoragePersistedState):
(WebKit::NetworkProcess::clearStorage): Deleted.
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::resetStoragePersistedState):
(WebKit::NetworkStorageManager::clearStorageForTesting): Deleted.
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.h:
* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.cpp:
(WKWebsiteDataStoreResetStoragePersistedState):
(WKWebsiteDataStoreClearStorage):
* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::resetStoragePersistedState):
(WebKit::NetworkProcessProxy::clearStorage): Deleted.
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resetStoragePersistedState):
(WebKit::WebsiteDataStore::clearStorage): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):
(WTR::TestController::resetStoragePersistedState):
* Tools/WebKitTestRunner/TestController.h: Rename clearStorage() to resetStoragePersistedState() to reflect its
actual use, and introduce a new clearStorage() function to clear multiple types of data at a time.

Canonical link: https://commits.webkit.org/258818@main
